### PR TITLE
Cloud 2832 tests for health validator 

### PR DESCRIPTION
**Pull request by maxres-fr**
This uses `mockery` and `testify/mock` to create mocks for the client interfaces. The following is how the mocks in the internal mocks package were generated. 

```
mockery --dir internal/k8s --output internal/mock --all
```

### CODE REVIEW

**Review by dgoldssfo — APPROVED** [review]
LGTM - I assume the testComments in TestConditionExpression are for internal consumption only

**Comment by maxres-fr** [comment]
> LGTM - I assume the testComments in TestConditionExpression are for internal consumption only

the comments are there to help someone understand later what failed.

File: cmd/doctor.go [hunk 1]
```diff
@@ -77,7 +76,6 @@ var operators = &cobra.Command{
 		return nil
 	},
 	DisableAutoGenTag: true,
-	SilenceErrors:     true, //We format and print errors ourselves during Execute().
```
Comment (jrcast) [inline cmd/doctor.go:80]: do you want to keep `SilenceUsage` out? Just checking.

**Review by jrcast — APPROVED** [review]

**Pull request merged**